pkg/agent/plugin/nodeattestor/k8s/k8scert.go

```diff
@@ -0,0 +1,173 @@
+package k8s
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	certificates "k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	csrPemBlockType          = "CERTIFICATE REQUEST"
+	ecPrivateKeyPemBlockType = "EC PRIVATE KEY"
+)
+
+func getKubeClient(kubeConfigFilePath, clientCertFilePath, clientKeyFilePath, caCertFilePath string) (*kubernetes.Clientset, error) {
+	if kubeConfigFilePath == "" {
+		// Try KUBECONFIG env variable
+		kubeConfigFilePath = os.Getenv("KUBECONFIG")
+		if kubeConfigFilePath == "" {
+			// Still no luck, try default (home)
+			home := os.Getenv("HOME")
+			if home != "" {
+				kubeConfigFilePath = path.Join(home, ".kube", "config")
+			}
+		}
+	}
+
+	if kubeConfigFilePath == "" {
+		return nil, fmt.Errorf("Unable to locate kubeconfig")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("Error accessing kubeconfig %s: %v", kubeConfigFilePath, err)
+	}
+
+	config.TLSClientConfig.CertFile = clientCertFilePath
+	config.TLSClientConfig.KeyFile = clientKeyFilePath
+	config.TLSClientConfig.CAFile = caCertFilePath
```

Comment:
if `~/.kube/config` already have the certs, we don't want to use them, instead sue the provided cert path.